packages/sdk/src/sdk/api/users/UsersApi.ts

```diff
@@ -54,6 +58,98 @@ export class UsersApi extends GeneratedUsersApi {
     this.logger = logger.createPrefixedLogger('[users-api]')
   }
 
+  /** @hidden
+   * Generate a new user id for use in creation flow
+   */
+  async _generateUserId() {
+    const response = new runtime.JSONApiResponse<{ data: string }>(
+      await this.request({
+        path: '/users/unclaimed_id',
+        method: 'GET',
+        headers: {},
+        query: {
+          noCache: Math.floor(Math.random() * 1000).toString()
+        }
+      })
+    )
+    return await response.value()
+  }
+
+  /** @hidden
+   * Create a user
+   */
+  async createUser(
+    params: CreateUserRequest,
+    advancedOptions?: AdvancedOptions
+  ) {
+    const { onProgress, profilePictureFile, coverArtFile, metadata } =
+      await parseParams('createUser', CreateUserSchema)(params)
+
+    const { data } = await this._generateUserId()
+    if (!data) {
+      throw new Error('Failed to generate userId')
+    }
+    const userId = HashId.parse(data)
+
+    const [profilePictureResp, coverArtResp] = await Promise.all([
+      profilePictureFile &&
+        retry3(
+          async () =>
+            await this.storage.uploadFile({
+              file: profilePictureFile,
+              onProgress,
+              template: 'img_square',
+              auth: this.auth
+            }),
+          (e) => {
+            this.logger.info('Retrying uploadProfilePicture', e)
+          }
+        ),
+      coverArtFile &&
+        retry3(
+          async () =>
+            await this.storage.uploadFile({
+              file: coverArtFile,
+              onProgress,
+              template: 'img_backdrop',
+              auth: this.auth
+            }),
+          (e) => {
+            this.logger.info('Retrying uploadProfileCoverArt', e)
+          }
+        )
+    ])
+
+    const updatedMetadata = {
+      ...metadata,
+      userId,
+      ...(profilePictureResp
+        ? { profilePictureSizes: profilePictureResp?.id }
+        : {}),
+      ...(coverArtResp ? { coverPhotoSizes: coverArtResp?.id } : {})
+    }
+
+    const entityMetadata = snakecaseKeys(updatedMetadata)
+
+    const cid = (await generateMetadataCidV1(entityMetadata)).toString()
+
+    // Write metadata to chain
+    const { blockHash, blockNumber } = await this.entityManager.manageEntity({
+      userId,
+      entityType: EntityType.USER,
+      entityId: userId,
+      action: Action.CREATE,
+      metadata: JSON.stringify({
+        cid,
```

Comment:
Do we still need `cid`? In TracksApi we just pass an empty string

Reply:
Yes. @raymondjacobson can give a better answer, but it's because of collectibles.

Reply:
should be gone soon ™️